src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedThrowMethod.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeGen;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Throws a System.ArgumentNullException with the given 'paramName'.
+    /// </summary>
+    internal sealed class SynthesizedThrowMethod : SynthesizedGlobalMethodSymbol
```

Comment:
>SynthesizedThrowMethod [](http://example.com/codeflow?start=26&length=22)

Is the separate "Throw" method necessary or could this be folded into "ThrowIfNull"?

Reply:
I was instructed to factor it in precisely this way. Will make sure to include the reference to the similar methods on ArgumentNullException in the runtime.￼￼

Reply:
Done. See recent changes in PEModuleBuilder.cs.

Reply:
I think(?) it's to do with inlining. IIRC the JIT wont inline a method with a `throw` in it. So `ThrowIfNull` now gets inlined on the fast path, and the overhead of the extra method call to `Throw` is negated by the fact that we only call it when we already know we're throwing an exception.